File: backend/auth-service/src/main/java/com/example/authservice/oauth/store/StoredAuthorization.java
package com.example.authservice.oauth.store;

import java.time.Instant;
import java.util.Set;
import lombok.Data;

/** Redis 에 저장하는 OAuth2Authorization 의 평면 형태. */
@Data
public class StoredAuthorization {
    private String id;
    private String clientId;
    private String principalName;
    private String grantType;
    private Set<String> scopes;
    /** OAuth2Authorization.getAttributes() 를 AS Jackson 모듈로 직렬화한 JSON */
    private String attributesJson;
    private StoredToken accessToken;
    private StoredToken refreshToken;
    private StoredToken idToken;

    @Data
    public static class StoredToken {
        private String value;
        private Instant issuedAt;
        private Instant expiresAt;
        private Set<String> scopes;
        private String metadataJson;
    }
}

File: backend/auth-service/src/main/kotlin/com/example/authservice/oauth/store/StoredAuthorization.kt
package com.example.authservice.oauth.store

import java.time.Instant

/** Redis 에 저장하는 OAuth2Authorization 의 평면 형태. */
class StoredAuthorization {
    var id: String? = null
    var clientId: String? = null
    var principalName: String? = null
    var grantType: String? = null
    var scopes: Set<String>? = null

    /** OAuth2Authorization.getAttributes() 를 AS Jackson 모듈로 직렬화한 JSON */
    var attributesJson: String? = null
    var accessToken: StoredToken? = null
    var refreshToken: StoredToken? = null
    var idToken: StoredToken? = null

    class StoredToken {
        var value: String? = null
        var issuedAt: Instant? = null
        var expiresAt: Instant? = null
        var scopes: Set<String>? = null
        var metadataJson: String? = null
    }
}
